daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf
+package value
+
+import scala.language.higherKinds
+import data.{Decimal, FrontStack, Ref, SortedLookupList, Time}
+import data.ImmArray.ImmArraySeq
+import iface.{PrimType => PT, Type, TypePrim}
+
+import scalaz.Id.Id
+import scalaz.syntax.traverse._
+import scalaz.std.option._
+import org.scalacheck.{Arbitrary, Gen, Shrink}
+import Arbitrary.arbitrary
+
+/** [[ValueGenerators]] produce untyped values; for example, if you use the list gen,
+  * you get a heterogeneous list.  The generation target here, on the other hand, is
+  * ''a pair of a type and a generator of values of that type''.  For example, you might
+  * generate the type `[Map Decimal]`, which would be accompanied by a `Gen` that produced
+  * [[Value]]s each guaranteed to be a list of maps, whose values are all guaranteed to
+  * be Decimals.
+  *
+  * As a user, you will probably be most interested in one of the generators derived
+  * from `genAddend`; if you need a generator in this theme not already supported by
+  * one such generator, you can probably derive a new one from `genAddend` yourself.
+  */
+object TypedValueGenerators {
+  sealed abstract class ValueAddend {
```

Comment:
The name "addend" refers to how values of the type will be summed with other values to form a description of possible Values.

Reply:
Projection